### PR TITLE
Support for Odroid C2 and Odrobian OS.

### PR DESCRIPTION
**Pull request by larmog**
The Odrobian image has multiarch enabled, supporting both amd64 and armhf.

### CODE REVIEW

**Comment by DorianGray** [comment]
Why not AUFS?

**Comment by larmog** [comment]
Yes why not? There's no xfs support anyway...
You can change storage driver in `k8s.conf`.

**Comment by DorianGray** [comment]
Why did you choose devmapper over aufs as the default for this build?

**Comment by larmog** [comment]
As I said, it's a matter of [choice](https://docs.docker.com/engine/userguide/storagedriver/selectadriver/#future-proofing). We use `rhel` at work and I have more experience with `devicemapper`. I categorise `kubernetes-on-arm` as a lab platform. The `loop-lvm` should not be used in production.

I started out using `aufs`, but changed before I created the PR, when I saw that @luxas used `devicemapper` for Banana Pi.

**Comment by luxas** [comment]
Thanks @larmog. Odroid C2 will not make it for `v0.7.0` (will release soon), but I'm really glad you and @DorianGray are working on it when you have the device.

As noticed in #74 
> I didn't know you used your own static compiled version of Docker. I use hypriotos and used your kubernetes-on-arm deb package.

Yes, I'm building docker statically for both ARMv6 and ARMv7 for Arch Linux and `systemd` OSes (.deb installations) that do not have docker (e.g. raspbian). That's the best way for now since one do not depend on pacman then, which caused some real issues for us with `1.9` (https://github.com/docker/docker/issues/18125). Please use it for now.

When you say multiarch system, I understand it as a 32-bit filesystem, and the arm64 core is obviously capable of running both 32-bit and 64-bit binaries.

`aufs` or `devicemapper` doesn't matter to me personally. It's configurable with `v0.7.0` anyway. (and it's the parallella that uses `devicemapper` btw)

Thanks for your work. I'll review it.

File: sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		apt-get install sudo -y
+	fi
+
+	echo "Installing dbus and avahi-daemon ..."
+	apt-get install -y dbus avahi-daemon
+}
+
+install_docker(){
+	apt-get install lxc lvm2 cgroup-tools apparmor bridge-utils -y
+	wget -O $DOCKER_HYPRIOT_DEB_FILE https://downloads.hypriot.com/$DOCKER_HYPRIOT_DEB
```
Comment (luxas) [inline sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh]: Please the statically linked daemon here instead.

File: sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		echo "Installing sudo ..."
+		apt-get install sudo -y
+	fi
+
+	echo "Installing dbus and avahi-daemon ..."
+	apt-get install -y dbus avahi-daemon
+}
+
+install_docker(){
+	apt-get install lxc lvm2 cgroup-tools apparmor bridge-utils -y
```
Comment (luxas) [inline sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh]: I think lxc isn't required here. not sure about lvm2 and cgroup-tools when you're using a statically linked daemon

File: sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+
+os_upgrade(){
+	echo "Upgrading packages ..."
+	apt-get update -y && apt-get dist-upgrade -y && apt-get upgrade -y
+}
+
+os_post_install(){
+	echo "Post install ..."
+	DEFAULT_HOSTNAME="odroid32"
```
Comment (luxas) [inline sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh]: nit: maybe `kube-odroid`?

**Comment by larmog** [comment]
The arm64 core i capable of  running 32-bit but you need to tell the OS that's what you want. On Debian for example: `dpkg --add-architecture armhf`, or else you get a runtime err. The Odrobian image used here allows for installing and running both `armhf` and `aarch64`. (I'm no expert, but done some tests)

File: sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh (hunk 3 above)
Comment (larmog) [inline sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh]: The name is default from the image creator. `32` means it allows `armhf`.

File: sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh (hunk 1 above)
Comment (larmog) [inline sdcard/rootfs/kube-systemd/etc/kubernetes/dynamic-env/os/odrobian.sh]: Ok. I made this PR before I realised that you use your own Docker build.

**Comment by larmog** [comment]
I'm not so sure you should merge this PR. Maybe it's better to stick with Arch and HypriotOS. Give it a thought and let me know, and I'll fix the PR.

**Comment by luxas** [comment]
I lean towards your Arch PR as the best alternative. Arch has been quite stable for RPi at least

**Comment by larmog** [comment]
Closing in favor of PR #80.

**Pull request closed without merging**